flask_hw/flask_rest/api/room/rooms.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from flask_restful import Resource, marshal_with, marshal
+
+from api.room.room_parsers import data_valid_for
+from api.room.structure import room_structure
+from db import Rooms, db
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class RoomsRes(Resource):
+
+    @marshal_with(room_structure)
+    def get(self, number=None):
+        room_all = Rooms.query.all()
+        header = {
+            "Cache-Control": "no-store, no-cache, must-revalidate",
+            "Pragma": "no-cache"
+        }
+        args = data_valid_for('GET')
+        logging.debug(room_all)
+        logging.debug(number)
+        logging.debug(args)
+        if number:
+            room = Rooms.query.get(number)
+            if room:
+                return room, header
+            else:
+                # todo: how to display message instead of Rooms
+                #  structure with nulls?
+                return {"message": "room not found"}, 404, header
+        else:
+            if args['status']:
+                rooms = Rooms.query.filter_by(status=args['status']).all()
+                return rooms, header
+            # todo: add if status not found?
+        return room_all, header
+
+    def post(self):
+        data = data_valid_for('POST')
+        logging.debug(data)
+        if not Rooms.query.get(data['number']):
+            new_room = Rooms(**data)
+            db.session.add(new_room)
+            db.session.commit()
+            location = {
+                "Location": f'/api/v0.1/rooms/{str(new_room.number)}'
+            }
+            return {"message": "room was added successfully"}, \
+                   201, location
+        msg = f"number {data['number']} already exists"
+        return {"message": msg}, 400
+
+    def put(self, number=None):
+        args = data_valid_for('PUT')
+        logging.debug(number)
+        logging.debug(args)
+        if number:
+            room = Rooms.query.get(number)
+            if room:
+                if args['number'] != number \
+                        and Rooms.query.get(args['number']):
+                    msg = f"number {args['number']} already exists"
+                    return {"message": msg}, 400
+                else:
+                    # why it's not work?
+                    # room = Rooms(**args)
```

Comment:
with this we create new room, and we need to update exist one

Reply:
thanks